### DISCUSSION TITLE
Multi-Simulation Error

### DISCUSSION THREAD

**Discussion by Husnain2244** [pyswmm/pyswmm#525] [Q&A]
Hi, I am facing a **Multi-Simulation Error** to run a demo example with a code, as given hereunder. I tried to figure out a solution, but I could not, I would be grateful for your suggestions. Thank you

```
from pyswmm import Simulation, Nodes, Links, SystemStats

with Simulation('C:/Users/husna/Downloads/EspressoMachine/espresso_machine.inp') as sim:
    system_stats = SystemStats(sim)
    for step in sim:
        print(sim.current_time)
```
**Error message:**

runfile('C:/Users/husna/Downloads/EspressoMachine/untitled3.py', wdir='C:/Users/husna/Downloads/EspressoMachine')
Traceback (most recent call last):

  File ~\anaconda3\Lib\site-packages\spyder_kernels\py3compat.py:356 in compat_exec
    exec(code, globals, locals)

  File c:\users\husna\downloads\espressomachine\untitled3.py:4
    with Simulation('C:/Users/husna/Downloads/EspressoMachine/espresso_machine.inp') as sim:

  File ~\anaconda3\Lib\site-packages\pyswmm\simulation.py:100 in __init__
    raise(MultiSimulationError("Multi-Simulation Error."))

MultiSimulationError: 
Multi-Simulation Error.
Due to architectural limitations inside of EPA-SWMM
multiple Simulations cannot be completed within a single
instance of Python.  Consider launching the model runs
using a Python Subprocess and a Queue manager.

**Set-up details:**
 - OS: Windows
 - Python Version: Python 3.11
 - PySWMM Version: 2.0

**Comment by Husnain2244** [comment]
> Hi, I am facing a **Multi-Simulation Error** to run a demo example with a code, as given hereunder. I tried to figure out a solution, but I could not, I would be grateful for your suggestions. Thank you
> 
> ```
> from pyswmm import Simulation, Nodes, Links, SystemStats
> 
> with Simulation('C:/Users/husna/Downloads/EspressoMachine/espresso_machine.inp') as sim:
>     system_stats = SystemStats(sim)
>     for step in sim:
>         print(sim.current_time)
> ```
> 
> **Error message:**
> 
> runfile('C:/Users/husna/Downloads/EspressoMachine/untitled3.py', wdir='C:/Users/husna/Downloads/EspressoMachine') Traceback (most recent call last):
> 
> File ~\anaconda3\Lib\site-packages\spyder_kernels\py3compat.py:356 in compat_exec exec(code, globals, locals)
> 
> File c:\users\husna\downloads\espressomachine\untitled3.py:4 with Simulation('C:/Users/husna/Downloads/EspressoMachine/espresso_machine.inp') as sim:
> 
> File ~\anaconda3\Lib\site-packages\pyswmm\simulation.py:100 in **init** raise(MultiSimulationError("Multi-Simulation Error."))
> 
> MultiSimulationError: Multi-Simulation Error. Due to architectural limitations inside of EPA-SWMM multiple Simulations cannot be completed within a single instance of Python. Consider launching the model runs using a Python Subprocess and a Queue manager.
> 
> **Set-up details:**
> 
> * OS: Windows
> * Python Version: Python 3.11
> * PySWMM Version: 2.0

The issue has been solved.

**Reply by CHENJie666666** [reply]
I have the same problem. Can you tell me how you solved it
